### PR TITLE
Update konstruktoid/ansible-role-hardening for cisagov's needs

### LINKED ISSUE #117: Enable `nodev` for `/home` partition, if present
# 💡 Summary #

We should ensure that `nodev` is enabled for the `/home` partition of this AMI.

## Motivation and context ##

This improves security.

### PR DESCRIPTION
**Pull request by bra1ncramp**
#  #

## 🗣 Description ##

Update the ansible-role-hardening-2 for cisagov's needs.

## 💭 Motivation and context ##

Our current ansible-role-hardening was forked from [konstruktoid/ansible-role-hardening](https://github.com/konstruktoid/ansible-role-hardening) a few years ago. Since then, significant changes and improvements have been made in the original code. 

When building out a new Ubuntu AMI, we found that our existing hardening role was not sufficiently hardening the system, so we forked the current code from [konstruktoid/ansible-role-hardening](https://github.com/konstruktoid/ansible-role-hardening) and edited it to meet our use cases.

Resolves cisagov/freeipa-server-packer#117.

## 🧪 Testing ##

We built out an Ubuntu server and applied this role to it. We were able to successfully access the server and pass Nessus compliance scans.

We have also [tested on a Debian server](https://github.com/cisagov/openvpn-packer/pull/93) as well as [a Fedora server](https://github.com/cisagov/freeipa-server-packer/pull/113).

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.

### CODE REVIEW

**Comment by jsf9k** [comment]
- [I verified that using these changes results in an OpenVPN AMI that functions as expected](https://github.com/cisagov/openvpn-packer/pull/93).
- [These changes uninstall `freeipa-server` and dependencies from the FreeIPA AMI](https://github.com/cisagov/freeipa-server-packer/pull/113) for some reason I've yet to skry.
- [I verified that using these changes results in a FreeIPA AMI that functions as expected _if I first reinstall `freeipa-server`_](https://github.com/cisagov/freeipa-server-packer/pull/113).

**Review by jsf9k — CHANGES_REQUESTED** [review]
In general, you should comment your deviations from [konstructoid/ansible-role-hardening](https://github.com/konstructoid/ansible-role-hardening) stating why they were necessary.

File: tasks/auditd.yml
Comment (jsf9k) [inline tasks/auditd.yml:16]: Why is this necessary?

File: tasks/compilers.yml
Comment (jsf9k) [inline tasks/compilers.yml:40]: Why was this added?

File: tasks/ipv6.yml
Comment (jsf9k) [inline tasks/ipv6.yml:16]: Why was this added?

File: tasks/timesyncd.yml
Comment (jsf9k) [inline tasks/timesyncd.yml]: Why was this added?

File: tasks/timesyncd.yml
Comment (jsf9k) [inline tasks/timesyncd.yml:20]: Why did this need to be added?

File: templates/etc/audit/rules.d/hardening.rules.j2
Comment (jsf9k) [inline templates/etc/audit/rules.d/hardening.rules.j2:2]: Why was this removed?

**Review by jsf9k — CHANGES_REQUESTED** [review]
I called out several places where I think a comment is really necessary to explain why lines were added or removed.

**Comment by jsf9k** [comment]
> * [I verified that using these changes results in an OpenVPN AMI that functions as expected](https://github.com/cisagov/openvpn-packer/pull/93).
> * [These changes uninstall `freeipa-server` and dependencies from the FreeIPA AMI](https://github.com/cisagov/freeipa-server-packer/pull/113) for some reason I've yet to skry.
> * [I verified that using these changes results in a FreeIPA AMI that functions as expected _if I first reinstall `freeipa-server`_](https://github.com/cisagov/freeipa-server-packer/pull/113).

Update:
* [I verified that using these changes results in an OpenVPN AMI that functions as expected](https://github.com/cisagov/openvpn-packer/pull/93).
* ~[These changes uninstall `freeipa-server` and dependencies from the FreeIPA AMI](https://github.com/cisagov/freeipa-server-packer/pull/113) for some reason I've yet to skry.~
* [I verified that using these changes results in a FreeIPA AMI that functions as expected ~_if I first reinstall `freeipa-server`_~](https://github.com/cisagov/freeipa-server-packer/pull/113).

File: tasks/timesyncd.yml
Comment (jsf9k) [reply tasks/timesyncd.yml:20]: I see this was added in 573b13ff3951526b98100e4d7c3eccd0379c5907 so that the modifications to the `timesyncd` configuration file are only made if the configuration file actually exists.  Can you add a comment to that effect?

File: tasks/timesyncd.yml
Comment (jsf9k) [reply tasks/timesyncd.yml]: Why is this change part of commit 573b13ff3951526b98100e4d7c3eccd0379c5907?  The commit message for that particular commit doesn't mention anything that explains this change.

File: tasks/auditd.yml
Comment (jsf9k) [reply tasks/auditd.yml:16]: I see that this was done as part of commit 452e5c9aecf680b6436ccca439f857173c691cdf, where you are updating `grub` anytime the `grub` configuration is modified.  Can you add a comment to that effect?

File: tasks/ipv6.yml
Comment (jsf9k) [reply tasks/ipv6.yml:16]: I see that this was done as part of commit 452e5c9aecf680b6436ccca439f857173c691cdf, where you are updating `grub` anytime the `grub` configuration is modified.  Can you add a comment to that effect?

File: tasks/compilers.yml
Comment (jsf9k) [reply tasks/compilers.yml:40]: It looks like this was added in commit c8bb4769a805b8e9cc5dbc817e845bdee63c591b to further lock down compilers.  Can you add a comment explaining why that is a good idea?

File: tasks/ipv6.yml
Comment (jsf9k) [reply tasks/ipv6.yml:16]: Taken care of in commit 6e19020.

File: tasks/auditd.yml
Comment (jsf9k) [reply tasks/auditd.yml:16]: Taken care of in commit 6e19020.

File: tasks/timesyncd.yml
Comment (jsf9k) [reply tasks/timesyncd.yml:20]: Taken care of in commit 76e0283.

**Comment by jsf9k** [comment]
@bra1ncramp - I resolved the conversations I could take care of by myself.  You'll have to take care of the rest before we can merge this.

File: tasks/compilers.yml
Comment (bra1ncramp) [reply tasks/compilers.yml:40]: I have continued to backport commits from upstream. See https://github.com/cisagov/ansible-role-hardening-2/commit/0ccbf12f1d214c61abd77db5baf5ede724c3476e

File: tasks/timesyncd.yml
Comment (bra1ncramp) [reply tasks/timesyncd.yml]: Per this commit, timesyncd does not exist on Fedora
https://github.com/konstruktoid/ansible-role-hardening/commit/e238a9d2be1a0ac8e4b9103e58456e2449d7129f

File: templates/etc/audit/rules.d/hardening.rules.j2
Comment (bra1ncramp) [reply templates/etc/audit/rules.d/hardening.rules.j2:2]: This was backported from upstream commit: https://github.com/cisagov/ansible-role-hardening-2/commit/b3571d307ab670304eba4e9cf3886e1bda99a3db

File: tasks/compilers.yml
Comment (jsf9k) [reply tasks/compilers.yml:40]: OK, but for future reference the way to backport commits from upstream is to either:
- `git rebase` on top of the upstream
- `git cherry-pick` from upstream

You shouldn't recreate the commits yourself.

File: tasks/timesyncd.yml
Comment (jsf9k) [reply tasks/timesyncd.yml]: [The commit you referenced](https://github.com/konstruktoid/ansible-role-hardening/commit/e238a9d2be1a0ac8e4b9103e58456e2449d7129f) does not contain this change.  That commit also states that the file _doesn't exist by default_ on Fedora 35; it doesn't say that the package doesn't exist at all.  In any event, [Fedora 35 is EOL as of almost a year ago](https://docs.fedoraproject.org/en-US/releases/eol/).

Also, please see [this comment](https://github.com/cisagov/ansible-role-hardening-2/pull/1#discussion_r1388217368), as it applies here as well.

File: templates/etc/audit/rules.d/hardening.rules.j2
Comment (jsf9k) [reply templates/etc/audit/rules.d/hardening.rules.j2:2]: OK, but please see [this comment](https://github.com/cisagov/ansible-role-hardening-2/pull/1#discussion_r1388217368), as it applies here as well.

File: tasks/timesyncd.yml
Comment (jsf9k) [reply tasks/timesyncd.yml]: I think you should remove this line, in the interest of staying as true to the upstream as possible and therefore making the pulling in of upstream commits in the future as smooth as possible.

File: tasks/compilers.yml
Comment (bra1ncramp) [reply tasks/compilers.yml:40]: I'm always learning

File: tasks/timesyncd.yml
Comment (bra1ncramp) [reply tasks/timesyncd.yml]: Removed line in [b353fe0](https://github.com/cisagov/ansible-role-hardening-2/pull/1/commits/b353fe083b358cca07f3cc643adc6ce671121068)

Comment (jsf9k) [reply tasks/timesyncd.yml]: Thank you!

**Review by jsf9k — APPROVED** [review]
Thank you @bra1ncramp!  Please make sure to read [this comment](https://github.com/cisagov/ansible-role-hardening-2/pull/1/files#r1388217368) for future reference regarding pulling in upstream changes.

I think we can merge once we get an approving review from @dav3r.

File: tasks/aide.yml
Comment (dav3r) [inline tasks/aide.yml]: Remove unneeded capitalization in a comment:
```suggestion
    # - /usr/sbin/audispd # This is not required for CIS benchmarks, so we won't be installing audispd
```

File: README.md
Comment (dav3r) [inline README.md:334]: Why are these packages commented out?  Note: this goes for all of the commented-out packages in this file. There should be some explanation of why.

File: defaults/main/packages.yml
Comment (dav3r) [inline defaults/main/packages.yml:33]: Why are these packages commented out?  Note: this goes for all of the commented-out packages in this file.  There should be some explanation of why.

File: tasks/main.yml
Comment (dav3r) [inline tasks/main.yml:7]: The reason for each set of commented-out tasks should be explained via a comment.

**Review by dav3r — COMMENTED** [review]
I have some questions before I approve.  My apologies if I missed any previous discussions about this stuff.

File: README.md
Comment (bra1ncramp) [reply README.md:334]: See 94f7b7442d3c7ad4e858678561ac7fb081005155

File: defaults/main/packages.yml
Comment (bra1ncramp) [reply defaults/main/packages.yml:33]: see 94f7b7442d3c7ad4e858678561ac7fb081005155

File: tasks/main.yml
Comment (bra1ncramp) [reply tasks/main.yml:7]: I would ask for Dev to assist with this, then - as I am following this:
https://github.com/cisagov/ansible-role-hardening/blob/master/tasks/main.yml

Comment (dav3r) [reply tasks/main.yml:7]: @bra1ncramp I think you can just put something generic for each commented-out section like:
```
# These tasks are not currently needed for our use case
```

Comment (bra1ncramp) [reply tasks/main.yml:7]: see cb81d61417d3a3a7e19bce5096868e9a88b645db

**Review by dav3r — APPROVED** [review]
Thanks for sticking with this one @bra1ncramp - everything LGTM now, nice work! 👍

**Review by jsf9k — APPROVED** [review]

**Comment by jsf9k** [comment]
@bra1ncramp - Please don't merge this PR yet.  I am still testing the change from commit dd19901.

**Comment by jsf9k** [comment]
> @bra1ncramp - Please don't merge this PR yet. I am still testing the change from commit [dd19901](https://github.com/cisagov/ansible-role-hardening-2/commit/dd1990132a3092455039a78efbe49af172257453).

OK, I think this one is good to merge now.  It took several attempts to get the `nodev` jazz working, but commit 380b2b3 does the trick.

Do you agree that it's ready to merge, @bra1ncramp?  If so then either merge or let me know and I'll click the button.

**Comment by konstruktoid** [comment]
Hi and sorry for commenting like this.

I'm planning to release 2.0.0 due to breaking changes (see eg https://github.com/konstruktoid/ansible-role-hardening/blob/8f537cdba94604be52fe33c726ae1c9aa61a06a3/defaults/main/sshd.yml) in a week or so, and since you're the most active fork, I was wondering if there any changes you would like implemented?

**Comment by jsf9k** [comment]
> Hi and sorry for commenting like this.
> 
> I'm planning to release 2.0.0 due to breaking changes (see eg https://github.com/konstruktoid/ansible-role-hardening/blob/8f537cdba94604be52fe33c726ae1c9aa61a06a3/defaults/main/sshd.yml) in a week or so, and since you're the most active fork, I was wondering if there any changes you would like implemented?

Hi @konstruktoid!  I don't have any changes right now, but I may in the future.  I want to move away from this fork and instead use your role directly, which may entail adding a few role variables.  (See cisagov/freeipa-server-packer#118 and cisagov/openvpn-packer#96.)

**Comment by konstruktoid** [comment]
That's great to hear @jsf9k 

I'll take a look at the variables, and you're of course welcome to submit any PRs as well.

**Comment by konstruktoid** [comment]
It took a bit longer than a week...
v2.0.0-rc.1 released: https://github.com/konstruktoid/ansible-role-hardening/releases/tag/v2.0.0-rc.1

**Pull request merged**
